### PR TITLE
Re-enable sendPageViewPerformance, start/stopTrackPageView and remove duration from IPageViewTelemetry

### PR DESCRIPTION
**Pull request by barustum**
- I have also updated the E2E application that is checked in on barustum.visualstudio.com to add two buttons, startTrackPage and stopTrackPage
- Removed duration field from IPageViewTelemetry

### CODE REVIEW

**Comment by barustum** [comment]
Currently working on adding a couple of unit tests.

File: ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts [hunk 1]
```diff
@@ -63,7 +61,8 @@ export class ApplicationInsights implements IAppInsights, ITelemetryPlugin, IApp
     /**
      * Logs that a page or other item was viewed. 
      * @param IPageViewTelemetry The string you used as the name in startTrackPage. Defaults to the document title.
-     * @param customProperties Additional data used to filter events and metrics. Defaults to empty.
+     * @param customProperties Additional data used to filter events and metrics. Defaults to empty. If a user wants
+     *                         to provide a custom duration, it'll have to be in customProperties
```
Comment (jpiyali) [inline ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts]: add field name for customers here

File: ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts [hunk 2]
```diff
@@ -82,23 +81,77 @@ export class ApplicationInsights implements IAppInsights, ITelemetryPlugin, IApp
...
+                name = window.document && window.document.title || "";
+            }
+
+            if (typeof url !== "string") {
+                url = window.location && window.location.href || "";
+            }
+
+            this._pageTracking.stop(name, url, properties);
+
+            if (this.config.autoTrackPageVisitTime) {
```
Comment (jpiyali) [inline ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts]: is this being supported in the config read currently?

File: ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts [hunk 3]
```diff
@@ -190,19 +243,21 @@ export class ApplicationInsights implements IAppInsights, ITelemetryPlugin, IApp
...
-                uri: url,
-                duration: duration,
+                uri: url
             };
+
+            // duration must be a custom property in order for the collector to extract it
+            if (CoreUtils.isNullOrUndefined(properties)) {
+                properties = {};
+            }
+            properties["duration"] = duration;
```
Comment (jpiyali) [inline ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts]: move to const

File: ApplicationInsights/JavaScriptSDK/TelemetryItemCreator.ts [hunk 4]
```diff
@@ -1,51 +1,74 @@
...
+    /**
+     * Create a telemetry item that the 1DS channel understands
+     * @param item domain specific properties; part B
+     * @param baseType telemetry item type. ie PageViewData
+     * @param envelopeName name of the envelope. ie Microsoft.ApplicationInsights.<instrumentation key>.PageView
+     * @param customProperties user defined custom properties; part C
+     * @param systemProperties system properties that are added to the context; part A
+     * @returns ITelemetryItem that is sent to channel
+     */
+    create(item: IPageViewTelemetryInternal | PageViewPerformance,
```
Comment (jpiyali) [inline ApplicationInsights/JavaScriptSDK/TelemetryItemCreator.ts]: can we track two stories, one for serialization and other for inheriting off generated type from applicationinsights-home

File: ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts (hunk 1 above)
Comment (barustum) [reply ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts]: It's hard to see what is highlighted in GitHub pull requests :(. But what is the field name?

File: ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts (hunk 2 above)
Comment (barustum) [reply ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts]: This config is coming from the extension's config that is passed as part of the IConfiguration object. Is this what you mean? :)

**Pull request merged**